src/foundry/client/apps/sidebar/apps/compendium.d.mts

```diff
@@ -27,53 +31,60 @@ declare global {
      *   height: window.innerHeight - 100,
      *   top: 70,
      *   left: 120,
-     *   scrollY: [".directory-list"],
-     *   dragDrop: [{ dragSelector: ".directory-item", dropSelector: ".directory-list" }],
-     *   filters: [{inputSelector: 'input[name="search"]', contentSelector: ".directory-list"}]
+     *   popOut: true
      * });
      * ```
      */
     static override get defaultOptions(): ApplicationOptions;
 
+    override get id(): string;
+
     override get title(): string;
 
+    override get tabName(): string;
+
+    override get canCreateEntry(): boolean;
+
+    override get canCreateFolder(): boolean;
+
     /**
      * A convenience redirection back to the metadata object of the associated CompendiumCollection
      */
     get metadata(): this["collection"]["metadata"];
 
-    override getData(options?: Partial<Options>): MaybePromise<object>;
-
-    override close(options?: Application.CloseOptions): Promise<void>;
+    override initialize(): void;
 
-    override activateListeners(html: JQuery): void;
+    override render(
+      force?: boolean | undefined,
+      options?: Application.RenderOptions<ApplicationOptions> | undefined,
+    ): unknown;
 
-    /**
-     * Handle opening a single compendium entry by invoking the configured document class and its sheet
-     * @param event - The originating click event
-     */
-    protected _onClickEntry(event: JQuery.ClickEvent): void;
+    override getData(options?: Partial<Options>): Promise<object>;
 
-    protected override _onSearchFilter(event: KeyboardEvent, query: string, rgx: RegExp, html: HTMLElement): void;
+    protected override _entryAlreadyExists(entry: DirectoryMixinEntry): boolean;
 
-    protected override _canDragStart(selector: string): boolean;
+    protected override _createDroppedEntry(
+      entry: DirectoryMixinEntry,
+      folderId?: string | undefined,
+    ): Promise<DirectoryMixinEntry>;
 
-    protected override _canDragDrop(selector: string): boolean;
+    protected override _getEntryDragData(entryId: string): object;
 
-    protected override _onDragStart(event: DragEvent): void;
+    protected override _onCreateEntry(event: PointerEvent): Promise<void>;
 
     /**
+     * Handle clicks on a footer button
+     * @param event - The originating pointer event
      * @internal
      */
-    protected override _onDrop(event: DragEvent): void;
+    _onClickFooterButton(event: PointerEvent): AdventureExporter | void;
 
-    protected override _contextMenu(html: JQuery): void;
+    protected _getDocumentDragData(documentId: string): object;
```

Comment:
Add `override` modifier

Reply:
Correction: No need for `override` BUT change the return type from `object` to `{ type: string; uuid: string; }`

Reply:
Next function down for instance - `_getFolderDragData` - should I replace the `object` return type with `{ type: "Folder", uuid: string}`? The actual return of that function as implemented is:
```js
return {
  type: "Folder",
  uuid: folder.uuid
};
```
Also `_getEntryDragData` a few above, should it have the same `{ type: string; uuid: string; }`?

Reply:
I think that would be more helpful than not, yes.